Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -253,9 +253,9 @@ def fortran
   end
 
   # @private
-  def effective_arch
-    if Homebrew.args.build_bottle? && Homebrew.args.bottle_arch
-      Homebrew.args.bottle_arch.to_sym
+  def effective_arch(args)
```

Comment:
When is this called with `args` unset?

Reply:
There are some places in `Requirement`s, I think one place is in `java_dep_if_needed`, where there is no way to easily pass down the `args`. And in those cases I don't think the value of `effective_arch` is used.

Reply:
Cool, fine with me assuming it works as expected 👍